Core/Handlers/MessageHandler.php

```diff
@@ -381,9 +393,15 @@ public function onHandleException(
                 $this->deferExchangeMessage($retryExchangeEnvelope, $this->retryURI);
             } // If it's an exchange that is failing and it should not be retried later
             else {
+                $headers = $exchangeBackup->getIn()->getContext()->toArray();
+                if($headers && $this->callbackEndpoint && in_array('callback',$headers) && in_array('callbackMethod',$headers) &&  $headers['callback'] == true && $headers['callbackMethod']!='') {
+                    $callbackEnvelope = new CallbackExchangeEnvelope($exchangeBackup, $exception->getProcessingContext());
+                    $this->addCallbackHeadersToEnvelope($callbackEnvelope, $exception, $processor);
+                    $callbackExchange = new Exchange($callbackEnvelope);
+                    $this->callbackEndpoint->produce($callbackExchange);
```

Comment:
The content inside the if can be simplified using some part of the existing code
```
$callbackExchangeEnvelope = new CallbackExchangeEnvelope($exchangeBackup, $exception->getProcessingContext());
$this->addCallbackHeadersToEnvelope($callbackExchangeEnvelope, $exception, $processor);
this->deferExchangeMessage($callbackExchangeEnvelope, $this->callbackUri);
```